### PR TITLE
updater: Check if awaited instance matches current install 

### PR DESCRIPTION
**Pull request by derrod**
### Description

When checking for running OBS instance ensure the executable of an OBS process is actually residing the the directory we want to update.

Based on an old WIP commit: https://github.com/notr1ch/obs-studio/commit/95909ee25c3bbab9f6adc2a5c3795b0c115dd185

Contains and closes #8141 if merged first.

### Motivation and Context

Allow updating OBS even if some other `obs64.exe` is running (e.g. updating portable install while non-portable version is running).

Also selfishly, this means I won't have to use ShareX when recording my screen while working on updater updates :P

### How Has This Been Tested?

Verified portable install can be updated while existing OBS install is running

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/win-update/updater/updater.cpp [hunk 1]
```diff
@@ -531,7 +527,12 @@ static inline DWORD WaitIfOBS(DWORD id, const wchar_t *expected)
 	if (!proc.Valid())
 		return WAITIFOBS_WRONG_PROCESS;
 
-	if (!GetProcessImageFileName(proc, path, _countof(path)))
+	if (!GetModuleFileNameExW(proc, nullptr, path, _countof(path)))
```
Comment (notr1ch) [inline UI/win-update/updater/updater.cpp:530]: Perhaps `QueryFullProcessImageNameW` is better here?

Comment (derrod) [reply UI/win-update/updater/updater.cpp]: Either seems to work. Given the note in the MS Docs about `GetModuleFileName` being primarily intended for use in debuggers I've updated it to use `QueryFullProcessImageName` instead.

**Comment by derrod** [comment]
Rebased now that #8141 is merged. Should be good to merge™

**Review by RytoEX — APPROVED** [review]
@notr1ch tentatively approved off-thread. Will merge soon.

**Pull request merged**